src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -29,6 +29,34 @@ public function __construct(MessageBusInterface $decoratedBus)
      */
     public function dispatch($message)
     {
+        $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 7);
+
+        $file = $trace[0]['file'];
```

Comment:
Could you isolate that bit (with the `for` and the `compact`) in its own method?

Reply:
Done